src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadResolver.cs

```diff
@@ -50,7 +50,16 @@ public static WorkloadResolver Create(IWorkloadManifestProvider manifestProvider
                 workloadRootPaths = packRootEnvironmentVariable.Split(Path.PathSeparator).Select(path => (path, false)).Concat(workloadRootPaths).ToArray();
             }
 
-            return new WorkloadResolver(manifestProvider, workloadRootPaths, currentRuntimeIdentifiers);
+            try
+            {
+                return new WorkloadResolver(manifestProvider, workloadRootPaths, currentRuntimeIdentifiers);
+            }
+            catch (Exception e)
+            {
+                var ex = new Exception(e.Message, e.InnerException);
+                ex.Data.Add("CLI_User_Displayed_Exception", true);
+                throw ex;
```

Comment:
What does this end up doing to the exception that the user would see?

Reply:
I put this code into a console app:
```C#
var file = Environment.GetEnvironmentVariable("foo");
        try
        {
            if (file == null || file.Length > -1)
            {
                throw new FileNotFoundException("file not found exception");
            }
        }
        catch (Exception e)
        {
            var ex = new Exception(e.Message, e.InnerException);
            ex.Data.Add("CLI_USER_DISPLAYED_EXCEPTION", true);
            Console.WriteLine(ex.ToString());
        }
```

When it executed, it displayed this:
`System.Exception: file not found exception`

I'm not a huge fan of the fact that it wraps it in a new Exception, but my goal was to make the stack disappear, and that happened, and it kept the message.

Reply:
Hmm...I did my original testing in a console app, too, because I wasn't sure how exactly this worked. I just tried it in the wild, and it worked less well than I'd hoped. It's possible this needs something more complicated. Temporarily reverting to draft.